nonsoolmateClient/src/pages/payment/index.tsx

```diff
@@ -18,23 +18,40 @@ export default function Payment() {
 
   // -------- 쿠폰 로직
   const [isCouponOpen, setIsCouponOpen] = useState(false);
+  const [isSelctUnivOpen, setIsSelectUnivOpen] = useState(false);
+  const [isSucessOpen, setIsSucessOpen] = useState(false);
+  const [isRandomMatchOpen, setIsRandomMatchOpen] = useState(false);
+  const [isQuitOpen, setIsQuitOpen] = useState(false);
+
```

Comment:
```tsx
  const [modalStatus, setModalStatus] = useState({
    isCouponOpen: false,
    isSelectUnivOpen: false,
    isSuccessOpen: false,
    isRandomMatchOpen: false,
    isQuitOpen: false,
  })
```
위와 같이 모든 모달들을 한꺼번에 상태관리 해줄 수 있을 것 같습니다!

```tsx
  function toggleModal(modal: string, isOpen: boolean) {
    setModalStatus((prev) => ({
      ...prev,
      [modal]: isOpen,
    }));
  }
```
그리고 요런식으로 모달의 열리고 닫히는 핸들러 함수를 하나로 통일해서 어떤 모달인지 인자로 넘겨주면 코드가 더 깔끔해질 것 같아요 !

Reply:
코드가 훨씬 간결해졌네요! 아래처럼 바꿔보았습니다!
```suggestion
  const [modalStatus, setModalStatus] = useState({
    isCouponOpen: false,
    isSelectUnivOpen: false,
    isSuccessOpen: false,
    isRandomMatchOpen: false,
    isQuitOpen: false,
  });

  function changeModalStatus(modalName: keyof typeof modalStatus, openModal: boolean) {
    setModalStatus((prevState) => ({
      ...prevState,
      [modalName]: openModal,
    }));
  }

```